src/adaptations/device-layer/LwIP/WarmSupport.cpp

```diff
@@ -55,7 +58,10 @@ namespace Platform {
 
 WEAVE_ERROR Init(WarmFabricStateDelegate * inFabricStateDelegate)
 {
-    // Nothing to do.
+#if WARM_CONFIG_SUPPORT_THREAD_ROUTING
+    // Inform Warm of current state of thread router.
+    Warm::ThreadRoutingStateChange(kInterfaceStateUp);
```

Comment:
I don't think this should be hard-coded.   Even though WARM_CONFIG_SUPPORT_THREAD_ROUTING is 1, a device may not always be operating as a Thread router, based on configuration set by the application.  For example, the application could call `ThreadStackMgr().SetThreadDeviceType(kThreadDeviceType_FullEndDevice)` to force the device to be an end-device instead of a router.

I think the right place to call Warm::ThreadRoutingStateChange() is in [GenericConnectivityManagerImpl_Thread<ImplClass>::_OnPlatformEvent()](https://github.com/openweave/openweave-core/blob/47d46e76e643c67d226f866ad2f90bec5b49fc07/src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericConnectivityManagerImpl_Thread.ipp#L41). 

This method should watch for kThreadStateChange events where the RoleChanged flag == true.  Whenever the Thread role changes, the code should call ThreadStackMgrImpl().GetThreadDeviceType() to determine whether the device is currently operating as a router (kThreadDeviceType_Router) and then call Warm::ThreadRoutingStateChange() with kInterfaceStateUp or kInterfaceStateDown accordingly.